compiler/rustc_pattern_analysis/src/rustc.rs

```diff
@@ -895,6 +895,14 @@ impl<'p, 'tcx: 'p> TypeCx for RustcMatchCheckCtxt<'p, 'tcx> {
             errors::OverlappingRangeEndpoints { overlap: overlaps, range: pat_span },
         );
     }
+
+    fn complexity_exceeded(&self) -> Result<(), Self::Error> {
+        if let Some(match_span) = self.whole_match_span {
+            Err(self.tcx.dcx().span_err(match_span, "reached pattern complexity limit"))
+        } else {
+            Err(self.tcx.dcx().err("reached pattern complexity limit"))
```

Comment:
You can use `self.scrut_span` as fallback if `whole_match_span` isn't set

Reply:
Done.